example/sorting/example_sort_adjoint.f90

```diff
@@ -0,0 +1,15 @@
+program example_sort_adjoint
+  use stdlib_sorting, only: sort_adjoint
+  implicit none
+  integer, allocatable :: array(:)
+  real, allocatable :: adjoint(:)
+
+  array = [5, 4, 3, 1, 10, 4, 9]
```

Comment:
This is a great example @jvdp1. Could it be made into a test as well? The test would check that `array(i)>=array(i-1)`, and that `nint(adjoint(i),kind=${ik}$)==array(i)`